smart_open/smart_open_lib.py

```diff
@@ -611,7 +628,11 @@ def s3_iter_bucket(bucket, prefix='', accept_key=lambda key: True, key_limit=Non
         if key_limit is not None and key_no + 1 >= key_limit:
             # we were asked to output only a limited number of keys => we're done
             break
-    pool.terminate()
+
+    try:
```

Comment:
@jsphpl Can you fix this? Then we're ready to merge (the tests we can add later). Cheers!